### ISSUE TITLE
find_package(Vulkan REQUIRED) bug

### ISSUE THREAD

**Issue by doomtr666** [Microsoft/vcpkg#4430]
I'm trying to write a port for V-EZ which requires the Vulkan SDK.
The first line of the CMakeLists.txt is **find_package(Vulkan REQUIRED)** and it failed to find the SDK.
Outside of vcpkg it works.

**Comment by ras0219-msft** [comment]
Is it relying on environment variables? We clear out most environment variables before building in order to ensure reproducibility across different machines.

Ideally, we would package the Vulkan SDK directly inside vcpkg and use it as a dependency.

To debug issues like this, you can use the `--trace` option to cmake. This can be added as part of the `OPTIONS` of `vcpkg_configure_cmake()`.

**Comment by doomtr666** [comment]
Yes, it relies on VULKAN_SDK environment variable, and yes it is cleared.

Packaging the whole SDK is a quite complex task (even though some tools are
already packaged).
Maybe it's possible to build only vk headers and libs, with the correct
runtime installation left to the user ?
Or.. maybe not clearing VULKAN_SDK :D


Le mar. 9 oct. 2018 à 13:55, Robert Schumacher <notifications@github.com> a
écrit :

**Comment by MarkIanHolland** [comment]
You could also do something similar to the CUDA port? Just check its installed and if not, prompt the user to download an installer.

**Comment by doomtr666** [comment]
Yes, just have to find a way to find the SDK...
Thank you !




Le mar. 9 oct. 2018 à 22:06, Mark Ian Holland <notifications@github.com> a
écrit :

**Comment by MarkIanHolland** [comment]
Vcpkg has a white list of environment variables, but after adding vulkan_sdk I still had trouble getting find Vulkan to work.

**Comment by nicole9456** [comment]
The Vulkan port #[4664 ](https://github.com/Microsoft/vcpkg/issues/4430) has been merged into microsoft:master branch on Nov 15,  2018 and it works on my testing machine, so I closed the issue. 
Would you please re-open it if you still have any issue on it.

**Comment by doomtr666** [comment]
Great!
